### DISCUSSION TITLE
Livewire & Sweetalert2 Confirm on click event

### DISCUSSION THREAD

**Discussion by pixieaka** [livewire/livewire#1991] [1. Help] [answered]
**This work perfect**

`<button href="javascript:void(0)" wire:click="delete({{ $user }})" onclick="confirm() || event.stopImmediatePropagation()">Delete</button>`

**With native confirm dialog the click event wait until response from confirm dialog but with Sweetalert2 and same HTML**

```
function confirm() {
    swal.fire({
        title: 'Are you sure?'
        , text: "You won't be able to revert this!"
        , icon: 'warning'
        , showCancelButton: true
        , confirmButtonColor: '#d33'
        , cancelButtonColor: '#3085d6'
        , confirmButtonText: 'Yes, delete it!'
        , reverseButtons: true
    }).then((result) => {
        if (result.isConfirmed) {
            return true;
        }
    })
}
```

**It just get false it didn't run wire click at all in all situations because it didn't wait for user response.**

**Comment by joshhanley** [comment]
I would remove the wire:click completely and just try to add something like this after confirm

```
onclick="confirm() && @this.delete({{$user}})"
```

**Reply by pixieaka** [reply]
@joshhanley It's work with native browser confirm. but with Sweetalert2 It didn't work, but I don't know why.

**Reply by joshhanley** [reply]
Maybe the native browser confirm also fires preventDefault, which sweetalert2 isn't. Can you add `event.preventDefault()` along with stop propagation and see if that works?

**Reply by pixieaka** [reply]
@joshhanley Fixed in https://stackoverflow.com/questions/64735164/confirm-on-click-html-didnt-work-with-sweetalert2-but-its-work-with-native-bro

**Comment by pixieaka** [comment, marked as answer]
Fixed in https://stackoverflow.com/questions/64735164/confirm-on-click-html-didnt-work-with-sweetalert2-but-its-work-with-native-bro
